server/api.go

```diff
@@ -399,7 +399,45 @@ func (a *API) oauthRedirectHandler(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err = a.p.store.StoreUserInWhitelist(mmUserID); err != nil {
+	tx, err := a.p.store.BeginTx()
+	if err != nil {
+		a.p.API.LogError("Unable to begin transaction", "error", err.Error())
+		http.Error(w, "Something went wrong", http.StatusInternalServerError)
+		return
+	}
+
+	if err = a.p.store.LockWhitelist(tx); err != nil {
```

Comment:
What happens if this fails? the transaction doesn't get commited  or rollbacked here right? because the defer that commits the transaction is after this error handling, right?